app/endpoints/pdf_processing.py

```diff
@@ -0,0 +1,61 @@
+from fastapi import APIRouter, UploadFile, File, HTTPException, Form
+from typing import List
+from app.services.pdf_processor import process_pdf
+import json
+import logging
+
+router = APIRouter()
+
+@router.post("/process_pdf/")
+async def process_pdf_endpoint(file: UploadFile = File(...),
+                               names: str = Form(...)):
+    """
+    Endpoint to process a PDF and find names, their coordinates, and fuzzy match
+    scores.
+
+    Parameters:
+    file (UploadFile): The uploaded PDF file.
+    names (List[str]): A list of names to compare.
+
+    Returns:
+    dict: A dictionary containing the names found, their coordinates, and fuzzy
+    match scores.
+    """
+    logging.info("Received request to process PDF")
+    logging.info(f"Nombres originales: {names}")
+    if not file.filename.endswith('.pdf'):
+        logging.error("Invalid file format")
+        raise HTTPException(status_code=400,
+                            detail="Invalid file format. Only PDF files are accepted.")
+
+    try:
```

Comment:
Try to have each block of Try, Except in a different function.That will help you a lot in readability and maintainability